app/controllers/frontend_controller.rb

```diff
@@ -110,7 +110,7 @@ def pick_buyer_or_provider_layout
   end
 
   def all_services
-    @api_selector_services = current_user.accessible_services.includes(:account).to_a if current_account.try!(:provider?)
+    @api_selector_services = current_user.present? ? current_user.accessible_services.includes(:account) : []
```

Comment:
Well this will make the query on developer portal and I do not think it is needed at all. By the way doing the `before_filter` is not needed if you just use this method (memoizing it) and make it a helper

```ruby
 @api_selector_services ||= request.site_account.provider? ? current_user.accessible_services.includes(:account) : Service.none
```